src/services/contactService.ts

```diff
@@ -1,53 +1,153 @@
 import {RemoveContactArgs} from '@sphereon/ssi-sdk.contact-manager';
-import {Party, Identity} from '@sphereon/ssi-sdk.data-store';
-import Debug, {Debugger} from 'debug';
-
-import {APP_ID} from '../@config/constants';
-import {cmAddContact, cmAddIdentity, cmGetContacts, cmRemoveContact, cmUpdateContact} from '../agent';
+import {Identity, Party} from '@sphereon/ssi-sdk.data-store';
+import {ActionType, EventLogger, EventLoggerBuilder, InitiatorType, LoggingEventType, LogLevel, SubSystem, System} from '@sphereon/ssi-sdk.core';
+import {agentContext, cmAddContact, cmAddIdentity, cmGetContacts, cmRemoveContact, cmUpdateContact} from '../agent';
 import {IAddIdentityArgs, ICreateContactArgs, IGetContactsArgs, IUpdateContactArgs} from '../types';
 
-const debug: Debugger = Debug(`${APP_ID}:contactService`);
+const logger: EventLogger = new EventLoggerBuilder()
+  .withContext(agentContext)
+  .withLogLevel(LogLevel.INFO)
+  .withSystem(System.CONTACT)
+  .withSubSystem(SubSystem.CONTACT_MANAGER)
+  .withInitiatorType(InitiatorType.SYSTEM)
+  .build();
 
 export const getContacts = async (args?: IGetContactsArgs): Promise<Array<Party>> => {
-  debug(`getContacts(${JSON.stringify(args)})...`);
+  await logger.logEvent({
+    type: LoggingEventType.AUDIT,
+    data: {
+      level: LogLevel.TRACE,
+      description: 'getContacts function call',
+      actionType: ActionType.READ,
+      actionSubType: 'get contacts',
+      diagnosticData: args,
+    },
+  });
+
   return await cmGetContacts(args);
 };
 
 export const createContact = async (args: ICreateContactArgs): Promise<Party> => {
-  debug(`createContact(${JSON.stringify(args)})...`);
+  await logger.logEvent({
+    type: LoggingEventType.AUDIT,
+    data: {
+      level: LogLevel.TRACE,
+      description: 'createContact function call',
+      actionType: ActionType.CREATE,
+      actionSubType: 'create contact',
+      initiatorType: InitiatorType.USER,
+      diagnosticData: args,
+    },
+  });
+
   return cmAddContact(args)
     .then((contact: Party) => {
-      debug(`createContact(${JSON.stringify(args)}) succeeded`);
+      logger.logEvent({
+        type: LoggingEventType.AUDIT,
+        data: {
+          description: 'contact created',
```

Comment:
Wise to include the display name/alias in these calls

Reply:
added the display name